site/src/components/TemplateLayout/TemplateLayout.tsx

```diff
@@ -57,10 +57,14 @@ export const TemplateLayout: FC<{ children?: JSX.Element }> = ({
       organizationId,
     },
   })
-  const { template, permissions: templatePermissions } = templateState.context
+  const {
+    template,
+    permissions: templatePermissions,
+    templateVersionVariables,
+  } = templateState.context
   const permissions = usePermissions()
 
-  if (!template || !templatePermissions) {
```

Comment:
I think those only should be loaded in the page form. You can also use `react-query` for them. Way easier IMO. But not a blocker.